File: projects/project_2/javascript_solution/public/js/file_parser.js
/*
* Responsible for parsing a line in a file
*
*/
function parseHeader(line) {
  //Do nothing for now
}

function parseLine(line) {
  chunks = line.split(" ");
  h = {};
  h["process"] = chunks[0];
  h["action"] = chunks[1];
  h["resource"] = chunks[2];

  return h;
}

File: projects/project_2/jruby_solution/file_parser.rb
module FileParser

  def parse_header(line)
    return line.split(" ").first.to_i
  end

  def parse_line(line)
    split_line = line.split(" ")
    hash = {}

    hash[:process_name] = split_line.shift
    hash[:action] = split_line.shift
    hash[:resource_name] = split_line.shift

    return hash
  end
end
